Automattic/studio
# Fix: Cannot create BrowserWindow before app is ready

Related to https://github.com/Automattic/dotcom-forge/issues/6878

## Proposed Changes

This PR adds additional `app.isReady()` check to `app.on( 'activate')` to ensure that the app is ready before it attempts to create a browser window. 

## Testing Instructions

* Pull the changes from this branch locally
* Run `nvm use && npm install && npm start` to start the app
* Press `cmd + W` on macOS to close the app window but keep the app running
* Open the app again 
* Confirm there are no errors in the terminal

**Notes**
I spent some time trying to trigger the error manually but with no luck. It seems that the steps that a user might take to trigger this error would be (since `activate` event is specifically designed for macOS, it would need to be tested on Mac):

* Run the application.
* Close the main window but do not quit the application. (Typically CMD + W on macOS)
* Click the application icon in the dock to reactivate it.

- I also tried simulating the `activate` event starting early before the app is ready by adding something like `app.emit( 'activate' )` early after app initialization but that also did not trigger the error on my end;
- I am assuming that the fix would work as `app.isReady()` would need to be `true` first before the BrowserWindow is created. It seems that the `activate` event can possible fire ( https://github.com/Automattic/dotcom-forge/issues/6878#issuecomment-2087334306 ) before the `ready` event so it makes sense to add this guard in that case. 

If you have any good suggestions on how to test this, let me know!

## Pre-merge Checklist
